singleheader/simdjson.cpp

```diff
@@ -698,14 +698,13 @@ really_inline bool add_overflow(uint64_t value1, uint64_t value2, uint64_t *resu
 }
 
 #ifdef _MSC_VER
-#pragma intrinsic(_umul128) // todo: this might fail under visual studio for ARM
+uint64_t umul128(uint64_t ab, uint64_t cd, uint64_t* hi);
```

Comment:
this should be marked really_inline. We want to be sure that it gets inlined.

Reply:
in this case the function is in different cpp files, so, no way to inline it as this would work only assuming all cpp files are amalgamated in specific order. In any case, your suggestion for _M_AMR64 is much better, and I'll be able to remove the extern completely.